### PR TITLE
config: Community Alerts System

### PR DESCRIPTION
**Pull request by prince-chrismc**
Config!

This is a POC I am looking for feedback on. This will allow contributors to "subscribe" to recipes which they are most invested in to stay informed.

Let me know if this is complicated enough

- I intended on limiting this further to open when PRs are opened as to not create spam
- some of the paths are wrong to match the full folders

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
As said in the other PR. I think this is great.
Do you think it would be possible to put the path<->user mapping in a file similar to [CODEOWNERS](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners), instead of "hardcoding" it in the wokflow ?

**Comment by prince-chrismc** [comment]
> Do you think it would be possible to put the path<->user mapping in a file similar to CODEOWNERS, instead of "hardcoding" it in the wokflow ?

Absolutely! but I don't want to invest that much time before proving out the idea. Once this is implement and if there's good engagement I could see the ROI for that 😄 

> Also, "alters" ? did you mean "alerts" ?

Now that I googled it.. yes I did 🙊

File: .github/workflows/alert-community.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        
+      - uses: ./.github/actions/alert-community
+        with:
+          files: "linter/*/*"
+          reviewers: "@ericLemanissier @prince-chrismc"
+          
+      - uses: ./.github/actions/alert-community
+        with:
+          files: "recipes/aaf/*/*"
+          reviewers: "@MartinDelille"
```
Comment (prince-chrismc) [inline .github/workflows/alert-community.yml:30]: @MartinDelille You previously expressed interest in following this recipe, let me know if this is okay to add you too :)

**Comment by RubenRBS** [comment]
My question with this would be, how would the user request to be added here? Afaik as a normal contributor the c3i bot won't let you go forwards with a PR for files under .github, so users wouldn't be able to actually add themselves to this file manually. 
What do you have in mind for that? I guess something like codeowners as @ericLemanissier suggested could work?

**Comment by prince-chrismc** [comment]
CODEOWNERS doesn't work because community members would need write permission to the repo.

But in a future version we can invest in make it easier and more dynamic...

I am a little doubtful many would be interested in this so I want to avoid doing work a head of time :)

**Comment by uilianries** [comment]
@RubenRBS Users can open PRs to the github folder, only the `.c3i` is protected. For instance, the PR https://github.com/conan-io/conan-center-index/pull/14912 was opened by Eric and is not blocked by the bot.

**Comment by uilianries** [comment]
@prince-chrismc The idea looks really great! Do you have any output checking your changes? As future feature, I would move those users and files to a YAML instead, so there is no risk of people breaking the CI due a typo or something, but as first step is awesome.

**Comment by prince-chrismc** [comment]
Of course I do https://github.com/prince-chrismc/conan-center-index/pull/54#issuecomment-1384889490

**Review by uilianries — APPROVED** [review]
As first step, it looks really promising! Thank you @prince-chrismc and @ericLemanissier !

File: .github/workflows/alert-community.yml (hunk 1 above)
Comment (MartinDelille) [reply .github/workflows/alert-community.yml:30]: Yes!

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,23 +0,0 @@
...
-# This feature is named 'CodeOwners', however, it's a Github's feature name.
-# On Conan Center Index, there are no owners/maintainers, all recipes are owned by the
-# community. This feature only helps people that are more interested in watching and following
-# specific recipes, instead of receiving a new notification for each new pull request.
-#
-# Full reference: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners
-#
-
-.github @ericLemanissier 
-recipes/aaf @MartinDelille
```
Comment (MartinDelille) [inline .github/CODEOWNERS:21]: I forgot I had this on and it explains why my github notification where totally wrong on the CCI! 😅

File: .github/workflows/alert-community.yml (hunk 1 above)
Comment (MartinDelille) [reply .github/workflows/alert-community.yml:30]: I'll be happy to have notification on more recipes (like ffmpeg). Can I specify a list here ?

File: .github/workflows/alert-community.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          
+      - uses: ./.github/actions/alert-community
+        with:
+          files: "recipes/gtk/*/*"
+          reviewers: "@ericLemanissier"
+          
+      - uses: ./.github/actions/alert-community
+        with:
+          files: "recipes/qt/*/*"
+          reviewers: "@ericLemanissier"
```
Comment (MartinDelille) [inline .github/workflows/alert-community.yml:40]: Another way would be to add a list of reviewers here.

**Review by MartinDelille — CHANGES_REQUESTED** [review]

**Review by SSE4 — APPROVED** [review]

File: .github/CODEOWNERS (hunk 2 above)
Comment (prince-chrismc) [reply .github/CODEOWNERS:21]: This never actually worked :) feel free to add any recipe you are interested in, I want to test this system out then maybe improve :)

**Comment by ericLemanissier** [comment]
@prince-chrismc houston we have a problem https://github.com/conan-io/conan-center-index/actions/runs/4037397219/jobs/6940663939 `Error: Unhandled error: HttpError: Resource not accessible by integration`
This is because the default token only has READ permissions https://github.com/conan-io/conan-center-index/actions/runs/4037397219/jobs/6940663939#step:1:17
It may be solved by adding
```
    permissions:
      pull-requests: write
      contents: write
      issues: write
```
(I've been bitten by the same problem in https://github.com/conan-io/conan-center-index/pull/15500, but I had no energy left to try to fix it, so I just dropped the message posting part)

**Comment by prince-chrismc** [comment]
You are absolutely correct , my mightmare is they have changed this several times in the past https://github.com/prince-chrismc/label-merge-conflicts-action#faq---how-do-i-fix-resource-not-accessible-by-integration 

The easy fix is just use `pull_request_target` and avoid all this

**Comment by Croydon** [comment]
Awesome!

Maybe we can do something similar for issues and their labels 💡

**Pull request merged**